src/syncerd/types.rs

```diff
@@ -209,46 +213,72 @@ impl From<Boolean> for bool {
     }
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-#[display(Debug)]
 pub struct WatchTransaction {
     pub id: TaskId,
     pub lifetime: u64,
-    #[serde(with = "hex")]
-    pub hash: Vec<u8>,
+    pub hash: Txid,
     pub confirmation_bound: u32,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
+impl fmt::Display for WatchTransaction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "WatchTransaction(id: {}, lifetime: {}, hash: {}, confirmations_bound: {})",
+            self.id, self.lifetime, self.hash, self.confirmation_bound
+        )
+    }
+}
+
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-#[display(Debug)]
 pub struct BroadcastTransaction {
     pub id: TaskId,
     #[serde(with = "hex")]
     pub tx: Vec<u8>,
     pub broadcast_after_height: Option<u64>,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
+impl fmt::Display for BroadcastTransaction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let bitcoin_tx_id =
+            bitcoin::Transaction::consensus_decode(std::io::Cursor::new(self.tx.clone()))
+                .map(|tx| tx.txid().to_string())
+                .unwrap_or_default();
+
+        write!(
+            f,
+            "BroadcastTransaction(id: {}, broadcast_after_height: {:?}, tx_id: {})",
+            self.id, self.broadcast_after_height, bitcoin_tx_id
+        )
+    }
+}
+
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-#[display(Debug)]
 pub struct GetTx {
     pub id: TaskId,
-    #[serde(with = "hex")]
-    pub hash: Vec<u8>,
+    pub hash: Txid,
+}
+
+impl fmt::Display for GetTx {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "GetTx(id: {}, hash: {})", self.id, self.hash,)
```

Comment:
`Display` derive is shorter here

Reply:
d1c3b39312a24e46f2fa8acd7df707ef454331e2